.github/workflows/required-tests.yml

```diff
@@ -0,0 +1,211 @@
+    pki-tests:
+      # This job depends on the 'build' job and waits till it completes.
+      # This job needs container to be started manually, as Github provided
+      # container **does not** use /usr/bin/init as its ENTRYPOINT.
+      name: Test PKI
+      needs: build
+      runs-on: ubuntu-latest
+      env:
+        CONTAINER: pkitest
+        BUILDDIR: /tmp/workdir
+        PKIDIR: /tmp/workdir/pki
+        LOGS: ${GITHUB_WORKSPACE}/logs.txt
+        COPR_REPO: "@pki/master"
+      strategy:
+        matrix:
+          os: ['30', '31']
+      steps:
+        - name: Clone the repository
+          uses: actions/checkout@v2
+
+        - name: Setup required test environment
+          run: IMAGE=fedora:${{ matrix.os }} travis/runner-init.sh
+
+        - name: Download PKI binaries from Build job
+          uses: actions/download-artifact@v1
+          with:
+            name: pki-build-${{ matrix.os }}
+
+        - name: Extract tar.gz for rpms
+          run: tar -xzf pki-build-${{ matrix.os }}/pki-rpms.tar.gz
+
+        - name: Install required packages
+          run: docker exec -i ${CONTAINER} dnf install -y 389-ds-base-legacy-tools findutils dnf-plugins-core sudo wget 389-ds-base
+
+        - name: Install PKI packages
+          run: docker exec -i ${CONTAINER} bash -c "find ${PKIDIR} -name '*.rpm' -and -not -name '*debuginfo*' | xargs dnf -y install"
+        
+        #- name: Enable PKI COPR repo
+        #  run: docker exec -i ${CONTAINER} dnf copr enable -y ${COPR_REPO}
+
+        - name: Install DS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ds-create.sh
+
+        - name: Install CA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ca-create.sh
+
+        - name: Install KRA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/kra-create.sh
+
+        - name: Install OCSP
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ocsp-create.sh
+
+        - name: Install TKS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tks-create.sh
+
+        - name: Install TPS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tps-create.sh
+
+        - name: Remove TPS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tps-remove.sh
+
+        - name: Remove TKS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/tks-remove.sh
+
+        - name: Remove OCSP
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ocsp-remove.sh
+
+        - name: Remove KRA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/kra-remove.sh
+
+        - name: Remove CA
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ca-remove.sh
+
+        - name: Remove DS
+          run: docker exec -i ${CONTAINER} ${PKIDIR}/travis/ds-remove.sh
+
+        - name: Gather journalctl logs
+          run: docker exec -i ${CONTAINER} bash -c "journalctl -u pki-tomcatd@pkitest > /var/log/pki/pki-journalctl.log"
+
+        - name: Compress logs
+          if: always()
```

Comment:
Just curious, what is the purpose of `if: always()`?

Reply:
`always()` makes sure that this step gets executed even if any of the previous step fails. By default, all the steps after a failed step are skipped. In future, we can change this to `failure()` which will run only when one of the previous step fails. [More information](https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#always)